chips/sam4l/src/gpio.rs

```diff
@@ -79,6 +80,8 @@ pub enum PeripheralFunction {
 const BASE_ADDRESS: usize = 0x400E1000;
 const SIZE: usize = 0x200;
 
+pub static INTERRUPT_COUNT: AtomicUsize = AtomicUsize::new(0);
```

Comment:
It's basically to avoid `unsafe` despite it being a global variable. Ideally it would be a `Cell`, but that's not `Sync`, so can't be a global. `AtomicUsize` is... Would it alleviate your concerns if I validated that it wasn't actually using atomic instructions (I've assumed since we're using the `Relaxed` ordering and don't use the result, LLVM will just use regular load/stores)?